### ISSUE TITLE
Getting JSON of whole database schema back

### ISSUE THREAD

**Issue by jayhawkins** [mevdschee/php-crud-api#227]
I've followed the instructions for implementation, but everytime I try to hit the api for just a list of states (ie. http://nec.dev/api/states) I get a dump of the whole database schema. Am I missing a flag or something along the way. I am using MySQL 5.7 and PHP 7.1.

Using the rewrite option and even when I don't, I still get the whole schema.

**Comment by jayhawkins** [comment]
May already be answered elsewhere.

**Comment by jayhawkins** [comment]
Debian 8 & for PATH_INFO I was using $fastcgi_path_info but it was always blank, so I changed it to $request_uri and in my api.php script, I parse the $_SERVER['PATH_INFO'] to take out the '/api' so I can get the list of states  ie. /api/states

It does however hamper me from using querystring directives in the url, so that's another issue. I think it's all related to the rewrite.

fastcgi_param  QUERY_STRING       $query_string;
fastcgi_param  REQUEST_METHOD     $request_method;
fastcgi_param  CONTENT_TYPE       $content_type;
fastcgi_param  CONTENT_LENGTH     $content_length;

fastcgi_param  SCRIPT_NAME        $fastcgi_script_name;
fastcgi_param  PATH_INFO          $request_uri;
#fastcgi_param  PATH_TRANSLATED    $document_root$fastcgi_path_info;
fastcgi_param  REQUEST_URI        $request_uri;
fastcgi_param  DOCUMENT_URI       $document_uri;
fastcgi_param  DOCUMENT_ROOT      $document_root;
fastcgi_param  SERVER_PROTOCOL    $server_protocol;
fastcgi_param  HTTPS              $https if_not_empty;

fastcgi_param  GATEWAY_INTERFACE  CGI/1.1;
fastcgi_param  SERVER_SOFTWARE    nginx/$nginx_version;

fastcgi_param  REMOTE_ADDR        $remote_addr;
fastcgi_param  REMOTE_PORT        $remote_port;
fastcgi_param  SERVER_ADDR        $server_addr;
fastcgi_param  SERVER_PORT        $server_port;
fastcgi_param  SERVER_NAME        $server_name;

# PHP only, required if PHP was built with --enable-force-cgi-redirect
fastcgi_param  REDIRECT_STATUS    200;

**Comment by mevdschee** [comment]
I'm installing a Debian 8 in a VM to see if I can reproduce your issue. Please be patient.

**Comment by mevdschee** [comment]
I have installed Debian 8. It looks like it has PHP 5. How did you install PHP? Which packages and repositories did you use?

**Comment by jayhawkins** [comment]
I removed PHP5 and installed PHP7. I followed some instructions off a page I found, and at this point, can't remember what repo I used. I have Nginx 1.6, PHP 7.1 and MySQL 5.7. I'm building an admin app for our client, but we will also have a mobile app interacting with the db too and I want to utilize your api script for that. But, at this point, any query string attached to the url gets dropped or something, because all I can do is like a /states and get a list, and /states/1 and get the state with an id of 1, but the filtering querystring gets dropped.

**Comment by jayhawkins** [comment]
So, in the fastcgi_params file I'm including in my nginx conf file for the site, I'm using query_string for path_info. I'm not strong on understanding the rewrite config, but I'm assuming that is my problem. 

This works fine:

http://nec2.dev/api/states/36

array (
'USER' => 'www-data', 'HOME' => '/var/www', 'HTTP_CONNECTION' => 'keep-alive', 'HTTP_COOKIE' => 'PHPSESSID=3forkl63q9r0sdeq8msig1c8nk', 'HTTP_ACCEPT_ENCODING' => 'gzip, deflate', 'HTTP_ACCEPT_LANGUAGE' => 'en-US,en;q=0.5', 'HTTP_ACCEPT' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8', 'HTTP_USER_AGENT' => 'Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0', 'HTTP_HOST' => 'nec2.dev', 'REDIRECT_STATUS' => '200', 'SERVER_NAME' => 'nec2.dev', 'SERVER_PORT' => '80', 'SERVER_ADDR' => '127.0.0.1', 'REMOTE_PORT' => '55788', 'REMOTE_ADDR' => '127.0.0.1', 'SERVER_SOFTWARE' => 'nginx/1.6.2', 'GATEWAY_INTERFACE' => 'CGI/1.1', 'SERVER_PROTOCOL' => 'HTTP/1.1', 'DOCUMENT_ROOT' => '/var/www/html/nec2', 'DOCUMENT_URI' => '/api.php', 'REQUEST_URI' => '/api/states/36', 'SCRIPT_NAME' => '/api.php', 'CONTENT_LENGTH' => '', 'CONTENT_TYPE' => '', 'REQUEST_METHOD' => 'GET', 'QUERY_STRING' => '/states/36', 'SCRIPT_FILENAME' => '/var/www/html/nec2/api.php', 'PATH_INFO' => '/states/36', 'FCGI_ROLE' => 'RESPONDER', 'PHP_SELF' => '/states/36', 'REQUEST_TIME_FLOAT' => 1490577758.772674, 'REQUEST_TIME' => 1490577758, ){"id":36,"abbreviation":"OH","name":"Ohio","createdAt":"2017-03-22 12:15:00","updatedAt":"2017-03-22 12:15:00"}

---------------------------------------------------------------------------------------------------------------
Here is my call and the resulting dump for what DOESN'T work:

http://nec2.dev/api/states?filter=name,eq,Ohio

array (
  'USER' => 'www-data',
  'HOME' => '/var/www',
  'HTTP_CACHE_CONTROL' => 'max-age=0',
  'HTTP_CONNECTION' => 'keep-alive',
  'HTTP_COOKIE' => 'PHPSESSID=3forkl63q9r0sdeq8msig1c8nk',
  'HTTP_ACCEPT_ENCODING' => 'gzip, deflate',
  'HTTP_ACCEPT_LANGUAGE' => 'en-US,en;q=0.5',
  'HTTP_ACCEPT' => 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8',
  'HTTP_USER_AGENT' => 'Mozilla/5.0 (X11; Linux x86_64; rv:45.0) Gecko/20100101 Firefox/45.0',
  'HTTP_HOST' => 'nec2.dev',
  'REDIRECT_STATUS' => '200',
  'SERVER_NAME' => 'nec2.dev',
  'SERVER_PORT' => '80',
  'SERVER_ADDR' => '127.0.0.1',
  'REMOTE_PORT' => '55736',
  'REMOTE_ADDR' => '127.0.0.1',
  'SERVER_SOFTWARE' => 'nginx/1.6.2',
  'GATEWAY_INTERFACE' => 'CGI/1.1',
  'SERVER_PROTOCOL' => 'HTTP/1.1',
  'DOCUMENT_ROOT' => '/var/www/html/nec2',
  'DOCUMENT_URI' => '/api.php',
  'REQUEST_URI' => '/api/states?filter=name,eq,Ohio',
  'SCRIPT_NAME' => '/api.php',
  'CONTENT_LENGTH' => '',
  'CONTENT_TYPE' => '',
  'REQUEST_METHOD' => 'GET',
  'QUERY_STRING' => '/states&filter=name,eq,Ohio',
  'SCRIPT_FILENAME' => '/var/www/html/nec2/api.php',
  'PATH_INFO' => '/states&filter=name,eq,Ohio',
  'FCGI_ROLE' => 'RESPONDER',
  'PHP_SELF' => '/states&filter=name,eq,Ohio',
  'REQUEST_TIME_FLOAT' => 1490576614.423613,
  'REQUEST_TIME' => 1490576614,
)Not found (entity)

**Comment by mevdschee** [comment]
Your 'QUERY_STRING' and 'PATH_INFO' variables seem unusual to me. I would expect them to be:

    'QUERY_STRING' => 'filter=name,eq,Ohio',
    'PATH_INFO' => '/states',

I'm not sure what is causing this.

**Comment by mevdschee** [comment]
Also.. I noted that the README states that the rewrite should be:

    rewrite ^/api(.*)$ /api.php$1 last;

and you use:

    rewrite ^/api(.*)$ /api.php?$1 last;

Wouldn't it simply help to follow the manual?

**Comment by jayhawkins** [comment]
You would be correct! Thank you so much for your patience and time. I can't tell you how many times it's been "1 character" that is the root of my troubles. 

A couple of other things I ran across:

I found some Nginx help where it was showing to add the PATH_INFO directive to the fastcgi_params file (which seems to be the file Nginx wants you to load now, instead of fastcgi.conf). DON'T do that.

Also, for some reason, I had to add:

fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;

to my Nginx site conf file. If I didn't, the PATH_INFO would be blank. Not sure why, as I stated, I'm not real familiar with rewrites yet. But just wanted you to know.

So here is the location part of my conf file for my nec2.dev site:

location ~ [^/]\.php(/|$) {
                fastcgi_split_path_info ^(.+\.php)(/.+)$;
                try_files $fastcgi_script_name =404;
                set $path_info $fastcgi_path_info;
                fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
                set $path_info $fastcgi_path_info;
                fastcgi_param PATH_INFO $path_info;
                fastcgi_index index.php;
                include fastcgi_params;
                fastcgi_pass 127.0.0.1:9000;
        }

Thanks again for all your help and for this script. This will make my application much easier to write.
